src/Shared/Microsoft.AspNetCore.Razor.Serialization.Json/StringCache.ExfiltratingEqualityComparer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Razor.Serialization.Json;
+
+internal sealed partial class StringCache
+{
+    /// <summary>
+    /// This is a gross hack to do a sneaky and get the value inside the HashSet out given the lack of any Get operations in netstandard2.0.
+    /// If we ever upgrade to 2.1 delete this and just use the built in TryGetValue method.
+    /// </summary>
+    /// <remarks>
+    /// This is fragile on the ordering of the values passed to the EqualityComparer by HashSet.
+    /// If that ever switches we have to react, if it becomes indeterminate we have to abandon this strategy.
+    /// </remarks>
+    private sealed class ExfiltratingEqualityComparer : IEqualityComparer<Entry>
```

Comment:
Can we just use `Dictionary<string, string>` instead?

Reply:
Honestly, this is code that pre-existed me working on Razor, so I haven't looked carefully at it. I'll take a closer look.

Reply:
I haven't realized the code existed, I guess I'm fine with it then.

Reply:
A `Dictionary<string, string>` would hold onto string instances indefinitely. I think it might be possible to transform this into a `Dictionary<int, WeakReference<string>>`, where the key is the hash code of the string. I'll take a look at that approach.

Reply:
Actually, no. That won't work because the hash codes would need to be unique. :smile:

Reply:
I spent a lot of time tinkering with the existing implementation and decided that it wasn't worth changing. I have a sneaking suspicion that it doesn't add much value, except in the Live Share scenarios, which I expect are limited. However, improving the implementation would mean extra testing for that scenario, which doesn't seem worth it for this pull request. 🤷